spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/PooledChannelConnectionFactory.java

```diff
@@ -221,6 +221,8 @@ private Channel createProxy(Channel channel, boolean transacted) {
 							return channel.confirmSelect();
 						case "isConfirmSelected":
 							return confirmSelected.get();
+						case "isPublisherConfirms":
```

Comment:
I don't think we need changes in this factory at all; the interface `default` method returns `false`.

Same thing with the `ThreadChannel...`

Reply:
@garyrussell I guessed the same.

but when I remove it, I get an `org.springframework.aop.AopInvocationException: AOP configuration seems to be invalid: tried calling method [public default boolean org.springframework.amqp.rabbit.connection.ChannelProxy.isPublisherConfirms()] on target [AMQChannel(amqp://guest@127.0.0.1:5672/,1)]; nested exception is java.lang.IllegalArgumentException: object is not an instance of declaring class`

or if I put only the `advisor.addMethodName("isPublisherConfirms");` and remove the `case "isPlublisherConfirms"`, I get a `org.springframework.aop.AopInvocationException: Null return value from advice does not match primitive return type for: public default boolean org.springframework.amqp.rabbit.connection.ChannelProxy.isPublisherConfirms()`

:/

Reply:
there is some spring-aop configuration I'm missing?

Reply:
Oh; ok; I guess it's a runtime JVM thing - the JVM has to check to see if the implementation has overridden the default method and, only then, call the default method if there is none; with a proxy he can't tell.

Sorry for the noise.